src/mjolnir/ingest_transit.cc

```diff
@@ -143,98 +129,104 @@ struct unique_transit_t {
+  };
 
   filesystem::recursive_directory_iterator gtfs_feed_itr(path);
   filesystem::recursive_directory_iterator end_file_itr;
   for (; gtfs_feed_itr != end_file_itr; ++gtfs_feed_itr) {
     if (filesystem::is_directory(gtfs_feed_itr->path())) {
       auto feed_path = gtfs_feed_itr->path().string();
-      LOG_INFO("Trying to load " + feed_path);
+      LOG_INFO("Loading " + feed_path);
       gtfs::Feed feed(feed_path);
       feed.read_feed();
-      LOG_INFO("Done Reading Feed");
+
       const auto& stops = feed.get_stops();
+      // 1st pass to add all the stations, so we can add stops to its children in a 2nd pass
+      for (const auto& stop : stops) {
+        if (stop.location_type == gtfs::StopLocationType::Station) {
+          auto& tile_info = get_tile_info(stop);
+          tile_info.stations.insert({stop.stop_id, feed_path});
+        }
+      }
 
+      // 2nd pass to add the platforms/stops
       for (const auto& stop : stops) {
-        if (stop.stop_id.empty()) {
+        // TODO: GenericNode & BoardingArea could be useful at some point
+        if (!(stop.location_type == gtfs::StopLocationType::StopOrPlatform) &&
+            !(stop.location_type == gtfs::StopLocationType::EntranceExit)) {
           continue;
         }
-        gtfs::Id currStopId = stop.stop_id;
-        float x = stop.stop_lon;
-        float y = stop.stop_lat;
 
-        GraphId currId =
-            GraphId(tile_level.tiles.TileId(y, x), TileHierarchy::GetTransitLevel().level, 0);
+        auto& tile_info = get_tile_info(stop);
 
-        tileTransitInfo currTile;
-
-        if (tile_map.find(currId) != tile_map.end()) {
-          currTile = tile_map[currId];
+        // if this station doesn't exist, we need to create it: we use the fact that this entry is
+        // not a station type to fake a station object in write_stops()
+        auto station_in_tile = tile_info.stations.find({stop.parent_station, feed_path});
+        if (station_in_tile != tile_info.stations.end()) {
+          tile_info.station_children.insert({{stop.parent_station, feed_path}, stop.stop_id});
         } else {
-          currTile.graphId = currId;
-        }
-
-        if (stop.location_type == gtfs::StopLocationType::StopOrPlatform ||
-            stop.location_type == gtfs::StopLocationType::EntranceExit) {
-          auto parent_in_tile = currTile.tile_stops.find({stop.parent_station, feed_path});
-          if (parent_in_tile != currTile.tile_stops.end()) {
-            currTile.tile_station_children.insert({{stop.parent_station, feed_path}, stop.stop_id});
-          } else {
-            currTile.tile_stops.insert({stop.stop_id, feed_path});
+          // we don't have the parent station, if
+          // 1) this stop has none or 2) its parent station is in another tile
+          // TODO: need to handle the 2nd case somehow!
+          auto parent_station = feed.get_stop(stop.parent_station);
```

Comment:
if the `parent_station` is empty we should not emit this warning and should skip this check entirely

Reply:
hm, then we'd need to do more changes as downstream logic currently relies on stops getting inserted into this `stations` map. I also just made this a bit more readable, didn't change too much of the overall logic.